README.md

```diff
@@ -358,6 +386,16 @@ someProperties.execute(() -> {
 // here the system properties are reverted
 ```
 
+We can also specify properties to delete from the default system properties:
+
+```java
+// when this object is active, some properties will be removed
+// from system properties
+SystemProperties someProperties = new SystemProperties()
+    .remove("property1")
+    .remove("property2");
```

Comment:
Hi @ashleyfrieze 

this does not compile due to the introduction of `SystemPropertiesImpl`.

Also this PR breaks backward compatibility. This was working before but no longer does:

```java
class SomeTest {
    @SystemStub
    private static SystemProperties systemProperties = new MySystemProperties().set("property", "value");
}

class MySystemProperties extends SystemProperties {

    public MySystemProperties() {
        super("key1", "value1");
        this.set("key2", "value2");
    }
}
```

Created #80 for this.